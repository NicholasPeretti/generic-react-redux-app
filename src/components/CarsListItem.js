import React from 'react'
import PropTypes from 'prop-types'
import CarsListItemDescription from './CarsListItemDescription'
import CarThumb from './CarThumb'
import Style from '../style'
import styled from 'styled-components'

const StyledArticle = styled.article`
  border: 2px solid ${Style.colors.gray};
  display: flex;
  flexDirection: row;
  margin: ${Style.spacing.padding1} 0px;
`

var CarsListItem = ({ car = {} }) => (
  <StyledArticle>
    <CarThumb url={car.pictureUrl} />
    <CarsListItemDescription car={car}/>
  </StyledArticle>
)

CarsListItem.propTypes = {
  car: PropTypes.shape({
    stockNumber: PropTypes.number.isRequired,
    manufacturerName: PropTypes.string.isRequired,
    modelName: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    mileage: PropTypes.shape({
      number: PropTypes.number.isRequired,
      unit: PropTypes.string.isRequired
    }).isRequired,
    fuelType: PropTypes.string.isRequired,
    pictureUrl: PropTypes.string.isRequired
  }).isRequired
}

export default CarsListItem
