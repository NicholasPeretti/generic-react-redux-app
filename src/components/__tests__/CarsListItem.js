import React from 'react'
import { shallow } from 'enzyme'
import CarsListItem from '../CarsListItem'
import PresentationalTest from '../../utils/tests/presentational'
import { MemoryRouter } from 'react-router-dom'

const WrappedCarsListItem = props => (
  <MemoryRouter>
    <CarsListItem {...props} />
  </MemoryRouter>
)

const props = {
  car: {
    stockNumber: 86987,
    manufacturerName: 'Fiat',
    modelName: 'Talento',
    color: 'white',
    mileage: {
      number: 102586,
      unit: 'km'
    },
    fuelType: 'Petrol',
    pictureUrl: 'http://localhost:3001/car.svg'
  }
}

describe('CarsListItem', () => {
  PresentationalTest(WrappedCarsListItem, props)

  const wrapper = shallow(<WrappedCarsListItem {...props} />)

  it('Should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
