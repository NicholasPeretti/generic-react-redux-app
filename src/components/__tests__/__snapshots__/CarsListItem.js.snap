// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarsListItem Should match snapshot 1`] = `"<article class=\\"sc-htpNat fkUnup\\"><img src=\\"http://localhost:3001/car.svg\\" style=\\"width:100px;height:80px;margin:12px\\"/><section class=\\"sc-bwzfXH deuhUd\\"><h3 style=\\"margin:0px;font-size:18px;font-weight:bold\\">Fiat Talento</h3><span style=\\"font-size:14px;font-weight:normal;margin:0px\\">Stock # 86987 - 102586 km - Petrol - white</span><span style=\\"font-size:14px;font-weight:normal;margin:0px\\"><a class=\\"sc-bdVaJa eBhWSy\\">View details</a></span></section></article>"`;
